emscripten/CMakeLists.txt

```diff
@@ -0,0 +1,174 @@
+cmake_minimum_required(VERSION 3.0.0)
+
+# ------------------------------------------------------------------------------
+
+SET(emscripten_root_path "$ENV{EMSCRIPTEN}")
+
+# Abort if not found. 
+if ("${emscripten_root_path}" STREQUAL "")
+  MESSAGE(FATAL_ERROR "Could not locate the Emscripten directory via the \
+                       EMSCRIPTEN environment variable! Set it up or pass \
+                       -Demscripten_root_path=xxx to CMake to specify the \
+                       directory.")
+endif()
+
+SET(CMAKE_TOOLCHAIN_FILE "${emscripten_root_path}/cmake/Modules/Platform/Emscripten.cmake")
+
+SET(project_name "libUncrustify")
+PROJECT(${project_name})
+
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+
+#get parent folder, add src
+GET_FILENAME_COMPONENT(unc_projdir ${CMAKE_CURRENT_LIST_DIR} DIRECTORY)
+
+# ------------------------------------------------------------------------------
+
+include(CheckCXXCompilerFlag)
+include(CheckIncludeFileCXX)
+include(CheckSymbolExists)
+include(CheckCXXSymbolExists)
+include(CheckTypeSize)
+
+#
+# Determine config
+#
+if(WIN32)
+  # Windows builds use src/windows_compat.h instead of config.h
+else()
+  # Generate config.h
+  set(avail_headers "")
+
+  set(headers
+    inttypes.h
+    memory.h
+    stdint.h
+    stdlib.h
+    strings.h
+    string.h
+    sys/stat.h
+    sys/types.h
+    unistd.h
+    utime.h
+  )
+  foreach(header ${headers})
+    string(TOUPPER "${header}" header_uc)
+    string(REGEX REPLACE "[^A-Z0-9_]" "_" include_var "HAVE_${header_uc}")
+    check_include_file_cxx("${header}" ${include_var})
+    if(${include_var})
+      list(APPEND avail_headers ${header})
+    endif()
+    unset(include_var)
+    unset(header_uc)
+  endforeach()
+  unset(headers)
+
+  check_include_file("stdbool.h" HAVE_STDBOOL_H)
+
+  set(symbols
+    memset
+    strcasecmp
+    strchr
+    strdup
+    strerror
+    strtol
+    strtoul
+  )
+  foreach(symbol ${symbols})
+    string(TOUPPER "${symbol}" symbol_uc)
+    string(REGEX REPLACE "[^A-Z0-9_]" "_" symbol_var "HAVE_${symbol_uc}")
+    check_cxx_symbol_exists("${symbol}" "${avail_headers}" ${symbol_var})
+    unset(symbol_var)
+    unset(symbol_uc)
+  endforeach()
+  unset(symbols)
+
+  unset(avail_headers)
+
+  check_type_size(_Bool _BOOL LANGUAGE C)
+
+  configure_file("${unc_projdir}/src/config.h.in" config.h @ONLY)
+endif()
+
+#
+# Generate uncrustify_version.h
+#
+set(PACKAGE_VERSION "0.64")
+
+find_package(PythonInterp REQUIRED)
+execute_process(
+  COMMAND ${PYTHON_EXECUTABLE} "${unc_projdir}/scripts/make_version.py"
+  WORKING_DIRECTORY ${unc_projdir}
+  RESULT_VARIABLE make_version_error
+  OUTPUT_VARIABLE make_version_output
+)
+
+if(${make_version_error})
```

Comment:
Can't we reuse some of the code from the main `CMakeFiles.txt` instead of duplicating code here?